packages/odyssey-react/src/components/Tab/index.tsx

```diff
@@ -0,0 +1,176 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { useState } from "react";
+import type { FunctionComponent, ReactElement, ReactText } from "react";
+
+export type PropsTabs = {
+  /**
+   * A collection of Tab.Panels to be rendered
+   */
+  children: ReactElement[],
+
+  /**
+   * The tab id attribute. Automatically generated if not provided.
+   */
+  id: string,
+
+  /**
+   * An aria label to be placed on the tab component. It
+   * should describe the purpose of the tabs.
+   */
+  ariaLabel: string,
+
+  /**
+   * Specifes which Tabs.Panel should be selected on mount
+   */
+  selectedId?: string,
+  
+  /**
+   * Callback when the selected tab is changed.
+   */
+  onTabChange?: (newId: string) => void
+}
+
+export type PropsTabsContainer = { 
+  children: ReactElement | ReactElement[],
+  id: string,
+  ariaLabel: string
+}
+
+export type PropsTabsPanelContainer = { 
+  children: ReactElement | ReactElement[]
+}
+
+export type PropsTabsPanel = { 
+  children: ReactText | ReactElement | ReactElement[],
+  label: string,
+  id?: string,
```

Comment:
hmmm this is tricky. If `id` is optional we break `a11y` and the use of `arial-labelledby` below.

one the other hand if we autogenerate IDs to solve for this we make the `defaultSelected` API unworkable 😅  We may need to revisit this to use index position or a declarative `isSelected` API